gtsfm/frontend/verifier/colmap_two_view_geometry_verifier.py

```diff
@@ -0,0 +1,195 @@
+
+
+MIN_INLIER_RATIO = 0.01
+MIN_NUM_TRIALS = 100000
+MAX_NUM_TRIALS = 1000000
+CONFIDENCE = 0.999999
+
+
+class ConfigurationType(Enum):
+    UNDEFINED = 0
+    # Degenerate configuration (e.g., no overlap or not enough inliers).
+    DEGENERATE = 1
+    # Essential matrix.
+    CALIBRATED = 2
+    # Fundamental matrix.
+    UNCALIBRATED = 3
+    # Homography, planar scene with baseline.
+    PLANAR = 4
+    # Homography, pure rotation without baseline.
+    PANORAMIC = 5
+    # Homography, planar or panoramic.
+    PLANAR_OR_PANORAMIC = 6
+    # Watermark, pure 2D translation in image borders.
+    WATERMARK = 7
+    # Multi-model configuration, i.e. the inlier matches result from multiple
+    # individual, non-degenerate configurations.
+    MULTIPLE = 8
+
+
+class ColmapTwoViewGeometryVerifier(VerifierBase):
+    def __init__(
+        self,
+        use_intrinsics_in_verification: bool,
+        estimation_threshold_px: float,
+    ) -> None:
+        """Initializes the verifier.
+
+        (See https://github.com/mihaidusmanu/pycolmap/blob/master/essential_matrix.cc#L98)
+
+        Args:
+            use_intrinsics_in_verification: Flag to perform keypoint normalization and compute the essential matrix
+                instead of fundamental matrix. This should be preferred when the exact intrinsics are known as opposed
+                to approximating them from exif data.
+            estimation_threshold_px: maximum distance (in pixels) to consider a match an inlier, under squared
+                Sampson distance.
+        """
+        self._use_intrinsics_in_verification = use_intrinsics_in_verification
+        self._estimation_threshold_px = estimation_threshold_px
+        self._min_matches = (
+            verifier_base.NUM_MATCHES_REQ_E_MATRIX
+            if self._use_intrinsics_in_verification
+            else verifier_base.NUM_MATCHES_REQ_F_MATRIX
+        )
+
+        # for failure, i2Ri1 = None, and i2Ui1 = None, and no verified correspondences, and inlier_ratio_est_model = 0
+        self._failure_result = (None, None, np.array([], dtype=np.uint64), 0.0)
+
+    def __estimate_two_view_geometry(
+        self,
+        uv_i1: np.ndarray,
+        uv_i2: np.ndarray,
+        camera_intrinsics_i1: Cal3Bundler,
+        camera_intrinsics_i2: Cal3Bundler,
+    ) -> Dict[str, Any]:
+        """Use the pycolmap Pybind wrapper to estimate an Essential matrix using LORANSAC.
+
+        Args:
+            uv_i1: array of shape (N3,2) representing coordinates of 2d points in image 1.
+            uv_i2: array of shape (N3,2) representing corresponding coordinates of 2d points in image 2.
+            camera_intrinsics_i1: intrinsics for image #i1.
+            camera_intrinsics_i2: intrinsics for image #i2.
+
+        Returns:
+            dictionary containing result status code, estimated relative pose (R,t), and inlier mask.
+        """
+
+        def get_pycolmap_camera_dict(camera_intrinsics: Cal3Bundler) -> pycolmap.Camera:
```

Comment:
if we are treating it as pycolmap.Camera, I dont know why there are so many _dict here. IIUC, its basically a dict, but that is an implementation detail that can be explain in comment without much change in variable names.

Reply:
Good point -- I'll rename to `get_pycolmap_camera()`?

Reply:
Yeah, please also refer to my other comment on moving it to a util file.

Reply:
Should be fixed now.